### PR TITLE
virsh_nodeinfo.py: Fix the type of cpus_nodeinfo, cpus_online, and cpus_total

### PR DESCRIPTION
**Pull request by mizumm**
type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option fails
because 'if cpus_nodeinfo != cpus_online:' statement in output_check()
is always true. That is because the type of cpus_nodeinfo is str, but the
type of cpus_online is byte.

To fix that, change the type of cpus_nodeinfo, cpus_online, and cpus_total
to int.

Signed-off-by: Masayoshi Mizuma <m.mizuma@jp.fujitsu.com>

### CODE REVIEW

**Comment by chunfuwen** [comment]
@mizumm , Could you upload your local test result ?

**Comment by mizumm** [comment]
@chunfuwen  Sure. here is the result.


- Before applying the patch
```
# avocado run --vt-type libvirt --vt-arch x86_64 --vt-machine-type q35 type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option
/usr/local/lib/python3.6/site-packages/avocado/plugins/run.py:287: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : 8502a221365175fd61abc0f78a7b0371722bee1b
JOB LOG    : /root/avocado/job-results/job-2020-03-30T08.15-8502a22/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option: FAIL: Virsh nodeinfo output didn't match number of CPU(s) (4.07 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 5.16 s
#
```

- After applying the patch
```
# avocado run --vt-type libvirt --vt-arch x86_64 --vt-machine-type q35 type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option
/usr/local/lib/python3.6/site-packages/avocado/plugins/run.py:287: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : 4b09ce05811ff991aaba37347d21ee38677763f2
JOB LOG    : /root/avocado/job-results/job-2020-03-30T08.26-4b09ce0/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option: PASS (5.84 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 6.93 s
# 
```

**Comment by mizumm** [comment]
@chunfuwen Could you check https://github.com/autotest/tp-libvirt/pull/2621#issuecomment-605971649 ?

File: libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py [hunk 1]
```diff
@@ -42,15 +42,20 @@ def output_check(nodeinfo_output):
         # Check number of CPUs, nodeinfo CPUs represent online threads in the
         # system, check all online cpus in sysfs
         cpus_nodeinfo = _check_nodeinfo(nodeinfo_output, "CPU(s)", 2)
-        cmd = "cat /sys/devices/system/cpu/cpu*/online | grep 1 | wc -l"
-        cpus_online = process.run(cmd, ignore_status=True,
-                                  shell=True).stdout.strip()
-        cmd = "cat /sys/devices/system/cpu/cpu*/online | wc -l"
-        cpus_total = process.run(cmd, ignore_status=True,
-                                 shell=True).stdout.strip()
+        cpus_nodeinfo = int(cpus_nodeinfo)
```
Comment (chloerh) [inline libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py:45]: How about `cpus_nodeinfo = int(_check_nodeinfo(nodeinfo_output, "CPU(s)", 2))`

**Comment by chloerh** [comment]
@Yingshun I think we have some concern about this particular case: 
`virsh.nodeinfo.no_option`  
Would you like to provide some additional info?

**Comment by Yingshun** [comment]
@mizumm  May I ask if above test result was executed on X86_64 arch host? As I know, the cpu socket in 'virsh nodeinfo' is incorrect as expected(there is a won't fix a bug).

File: libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py (hunk 1 above)
Comment (mizumm) [reply libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py]: Thank you for the suggestion, it's great! I pushed the change.

**Comment by mizumm** [comment]
@Yingshun Sorry for the delay. I was executed the test on x86_64 host.
Here is the log with  the latest tp-libvirt:

```
# avocado run --vt-type=libvirt type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option
JOB ID     : 80ca756b9a9a09a80bc24617ab244a1e3f186214
JOB LOG    : /root/avocado/job-results/job-2021-12-09T16.26-80ca756/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option: STARTED
 (1/1) type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option: FAIL: Virsh nodeinfo output didn't match number of CPU(s) (15.53 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 21.21 s
#
```

And the debug.log
```
[stdlog] 2021-12-09 16:26:23,754 stacktrace       L0039 ERROR|
[stdlog] 2021-12-09 16:26:23,754 stacktrace       L0041 ERROR| Reproduced traceback from: /usr/local/lib/python3.9/site-packages/avocado_vt/test.py:275
[stdlog] 2021-12-09 16:26:23,755 stacktrace       L0045 ERROR| Traceback (most recent call last):
[stdlog] 2021-12-09 16:26:23,755 stacktrace       L0045 ERROR|   File "/root/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-libvirt/libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py", line 200, in run
[stdlog] 2021-12-09 16:26:23,755 stacktrace       L0045 ERROR|     output_check(output)
[stdlog] 2021-12-09 16:26:23,755 stacktrace       L0045 ERROR|   File "/root/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-libvirt/libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py", line 68, in output_check
[stdlog] 2021-12-09 16:26:23,755 stacktrace       L0045 ERROR|     test.fail("Virsh nodeinfo output didn't match "
[stdlog] 2021-12-09 16:26:23,755 stacktrace       L0045 ERROR|   File "/usr/local/lib/python3.9/site-packages/avocado/core/test.py", line 979, in fail
[stdlog] 2021-12-09 16:26:23,755 stacktrace       L0045 ERROR|     raise exceptions.TestFail(message)
[stdlog] 2021-12-09 16:26:23,755 stacktrace       L0045 ERROR| avocado.core.exceptions.TestFail: Virsh nodeinfo output didn't match number of CPU(s)
[stdlog] 2021-12-09 16:26:23,755 stacktrace       L0046 ERROR|
```

After applied the patch, the test passes:
```
# avocado run --vt-type=libvirt type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option
JOB ID     : fe64f88a07d0f4492080d6480e616824115da269
JOB LOG    : /root/avocado/job-results/job-2021-12-09T17.25-fe64f88/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option: STARTED
 (1/1) type_specific.io-github-autotest-libvirt.virsh.nodeinfo.no_option: PASS (24.19 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 29.85 s
#
```

**Comment by chunfuwen** [comment]
close it since there are no any updates for long time.
If author find it is required, there is opportunity to open it again

**Pull request closed without merging**